python-libraries/data-platform-catalogue/data_platform_catalogue/client/exceptions.py

```diff
@@ -0,0 +1,38 @@
+class CatalogueError(Exception):
+    """
+    Base class for all errors.
+    """
+
+
+class ReferencedEntityMissing(CatalogueError):
+    """
+    A referenced entity (such as a user or tag) does not yet exist when
+    attempting to create a new metadata resource in the catalogue.
+    """
+
+
+class InvalidDomain(CatalogueError):
+    """
+    Exception thrown when a domain does not exist
+    """
+
+
+class MissingDatabaseMetadata(Exception):
+    """
+    Exception thrown when a entity is attempted to be ingested without
+    a given metadata specification
+    """
+
+
+class EntityDoesNotExist(CatalogueError):
+    """
+    Exception thrown when an entity that does not exist is trying to be
+    accessed in DataHub
+    """
```

Comment:
Exception name is sufficient
```suggestion
    pass
```